ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java

```diff
@@ -407,6 +426,7 @@ public void close() {
       try {
         getServerRpc().close();
       } catch(IOException ignored) {
+        // TODO(jiacheng): transition state to EXCEPTION here?
```

Comment:
In the existing code I don't think we transition to EXCEPTION very often. In many places like this, we might wanna transition the state and expose the exception to the outside. And the Ratis user can just observe the `RaftServer` lifecycle and choose to restart the RaftServer or do whatever else.

Reply:
When it is in CLOSING state, it must transit to CLOSED but won't transit to EXCEPTION.  CLOSING must handle the exceptions, if there are any.

Reply:
FYI this logic is gone tgt with the `ErrorRecorded` interface. I no longer have this default behavior in the `RaftServerProxy` because it now takes an `uncaughtExceptionHandler` from the application. Is that okay? @szetszwo